Objective-C/CBLDatabase.mm

```diff
@@ -950,102 +952,127 @@ - (BOOL) saveDocument: (CBLDocument*)document
 #pragma mark - RESOLVING REPLICATED CONFLICTS:
 
 
-- (bool) resolveConflictInDocument: (NSString*)docID error: (NSError**)outError {
-    CBL_LOCK(self) {
-        C4Transaction t(_c4db);
-        if (!t.begin())
-            return convertError(t.error(), outError);
+- (bool) resolveConflictInDocument: (NSString*)docID
+              withConflictResolver: (id<CBLConflictResolver>)conflictResolver
+                             error: (NSError**)outError {
+    while (true) {
+        CBLDocument* localDoc;
+        CBLDocument* remoteDoc;
+        CBL_LOCK(self) {
+            C4Transaction t(_c4db);
+            if (!t.begin())
+                return convertError(t.error(), outError);
+            
+            // Read local document:
+            localDoc = [[CBLDocument alloc] initWithDatabase: self documentID: docID
+                                              includeDeleted: YES error: outError];
+            if (!localDoc)
+                return false;
+            
+            // Read the conflicting remote revision:
+            remoteDoc = [[CBLDocument alloc] initWithDatabase: self documentID: docID
+                                               includeDeleted: YES error: outError];
+            if (!remoteDoc || ![remoteDoc selectConflictingRevision])
+                return false;
+            
+            if (!t.commit())
+                return convertError(t.error(), outError);
+        }
         
-        // Read local document:
-        CBLDocument* localDoc = [[CBLDocument alloc] initWithDatabase: self
-                                                           documentID: docID
-                                                       includeDeleted: YES
-                                                                error: outError];
-        if (!localDoc)
-            return false;
+        conflictResolver = conflictResolver ?: [CBLConflictResolution default];
         
-        // Read the conflicting remote revision:
-        CBLDocument* remoteDoc = [[CBLDocument alloc] initWithDatabase: self
-                                                            documentID: docID
-                                                        includeDeleted: YES
-                                                                 error: outError];
-        if (!remoteDoc || ![remoteDoc selectConflictingRevision])
-            return false;
+        Assert(localDoc == nil && remoteDoc == nil, @"Local and remote document shouldn't be empty \
+               at same time, when resolving conflict.");
+        localDoc = localDoc.isDeleted ? nil : localDoc;
+        remoteDoc = remoteDoc.isDeleted ? nil : remoteDoc;
```

Comment:
I think localDoc or remoteDoc here should be set to nil because the log messages in line 994 and the logic in line 1018 will not be correct. You can do this when creating CBLConflict directly.

Reply:
I have a typo in the original comment. Can you review check it again?